### ISSUE TITLE
[Hint & Question] : CycloneDDS now in conan-center

### ISSUE THREAD

**Issue by dotChris90** [eclipse-cyclonedds/cyclonedds#1415]
Hello Cyclone-DDS team, 

i just want to inform you guys that cyclonedds now become a conan package. 
see --> https://conan.io/center/cyclonedds

the current package just offers the C APIs with the options : with shared memory  (using iceoryx), with ssl (including openssl) and enable security. 

The current package not including the idlc and the corresponding cmake functions because this tools are for the build and not for target (in case of cross compilation the C APIs must be compiled for target but the idlc and so must be compiled with default profile and for the build system e.g. Ubuntu x86_64 or so). 

My question now is : are there any other important options we should add to the package or are they fine for start? 
Happy coding and thanks for your excellent work on cyclonedds!

**Comment by eboasson** [comment]
Hi @dotChris90, this is great news! Thank you so much for putting the effort in ☺️

I am pretty sure Windows users will appreciate the option matrix you have provided, I think on Linux there'll be much less value in being able to decide whether or not you OpenSSL has been linked in, because everyone's got it installed anyway. In short, you could probably reduce the matrix a little bit, but I am not sure that that would be worth the effort!

In short, it seems to me this is a great start.

As to the IDL compiler, perhaps it would be a possibility to create a separate "cyclonedds-idlc" package? The output of the IDL compiler is independent of platform and build options.

**Comment by dotChris90** [comment]
@eboasson Hi :D 

i also think cyclonedds-idlc or cyclonedds-tooling (in case you will add more build tools) is a good idea. 
Currently prepare package and  i guess i will push it in some weeks to conan-center. 

I just have to check if there exist currently new rules for "packages without libs / just executables". 
I will keep you updated ;) 

And btw nice work you did with the cyclonedds command line tools (the python CLI) - i never though a DDS implementation can have a so wonderful tooling!

**Comment by eboasson** [comment]
Thank you for the nice words! The credits really go to @thijsmie for figuring out how to do that in such a nice way ☺️.

**Comment by thijsmie** [comment]
Thanks for the praise 😄

**Comment by thijsmie** [comment]
Answer given on #1439.
